jujugui/static/gui/src/app/components/user-profile/model-list/model-list.js

```diff
@@ -177,6 +177,16 @@ YUI.add('user-profile-model-list', function() {
           return null;
         }
       }
+      let lastConnection = model.lastConnection || '--';
+      // Verify that whether we have a valid date; invalid dates will return
+      // NaN.
+      if (!isNaN(Date.parse(lastConnection))) {
```

Comment:
Instead of doing this check I'd prefer if the `DateDisplay` component was robust enough to handle invalid dates. This would mean that on line 204 below it would change to:
```JavaScript
<juju.components.DateDisplay
  date={model.lastConnection}
  relative={true} />
```